appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockChartDataModel.java

```diff
@@ -0,0 +1,222 @@
+        // Data Model in this loop
+        for (int i = 0; i < entries.length; i += tupleSize) {
+            /* (i, i + 1, ..., i + tupleSize - 1) forms the tuple
+               E.g.: i = 0, tupleSize = 2, (0, 1) indexes represent the tuple
+               Therefore if the last index is greater than the entries length,
+               the tuple group is invalid and the method should return. */
+            if (i + tupleSize - 1 >= entries.length) {
+                break;
+            }
+
+            // Create an array having the capacity to hold a tuple
+            String[] tuple = new String[tupleSize];
+
+            // Set the appropriate elements to the tuple
+            for (int j = 0; j < tupleSize; ++j) {
+                tuple[j] = entries[i + j];
+            }
+
+            // Add the tuple to the Data Model
+            addEntryFromTuple(tuple);
+        }
+
+        // Perform the defined action after data import
+        postDataImportAction();
+    }
+
+    /**
+     * Sets the default elements for the Data Model.
+     * To be called when no data is specified.
+     */
+    protected abstract void setDefaultElements();
+
+    /**
+     * Sets the default styling properties of the Data Series.
+     */
+    protected abstract void setDefaultStylingProperties();
+
+    /**
+     * Sets the elements of the Data Series from the specified columns.
+     *
+     * @param columns List of columns to import from
+     */
+    public void setElementsFromColumns(List<List<String>> columns) {
+        // Clear the current entries
+        clearEntries();
+
+        // Initially, the row size of the columns has to be determined
+        int rows = 0;
+
+        for (int i = 0; i < columns.size(); ++i) {
+            int columnSize = columns.get(i).size();
+
+            // Bigger column found; Update rows entry
+            if (columnSize > rows) {
+                rows = columnSize;
+            }
+        }
+
+        // Iterate through all the rows (except the first, which is the columnNames)
+        for (int i = 1; i < rows; ++i) {
+            // Create an array that can hold all the tuple entries
+            String[] tuple = new String[getTupleSize()];
+
+            // Iterate through all the columns.
+            // Column size is the same as the tuple size
+            for (int j = 0; j < getTupleSize(); ++j) {
+                List<String> column = columns.get(j);
+
+                if (column.size() == 0) { // Empty column; Use default value
+                    tuple[j] = getDefaultTupleEntry(i - 1); // Minus one to compensate for skipped entry
+                } else if (column.size() > i) { // Column has value; Use the column's value (if the column is large enough)
+                    tuple[j] = column.get(i);
+                } else {
+                    // Column too small; Use empty String (up for interpretation
+                    // for addEntryFromTuple)
+                    tuple[j] = "";
```

Comment:
@ewpatton I figured it would kind of skew the results. Consider the two columns:
* 1, 2, 3
* 10, 45

Now we combine these two. The first two entries are matched perfectly:
* (1, 10)
* (2, 45)

The third one has x = 3, but no y. We set y = 3 by the typical default value definition. Does the entry (3, 3) with y value = 3 really make sense? At this point it seems really arbitrary, so for me it made more sense to just set the entry as empty, and have the addEntryFromTuple just not parse that tuple (which is the behavior now)